Add student resources page

hirundinidae opened the pull request:
## [MED] Student Feature Component  
![image](https://user-images.githubusercontent.com/65516245/127262312-2198bc9f-4519-4b76-bd2a-b3e1229e0fc4.png)
![screenshot_20210728_000905](https://user-images.githubusercontent.com/65516245/127262322-45264155-caad-46e7-84d1-182f518e023b.png)
![screenshot_20210728_000845](https://user-images.githubusercontent.com/65516245/127262329-781d00df-5be3-4816-adf1-159f50b2dc22.png)

hirundinidae committed:
Copy component from ui library

hirundinidae committed:
Add student resources to resources page

hirundinidae committed:
Fix margins

hirundinidae committed:
Fix PropTypes for resource section

hirundinidae committed:
Change to use nextjs image and link

hirundinidae committed:
Fix links and text width

xsharonhe commented on components/StudentResource.jsx:
```diff
@@ -0,0 +1,124 @@
+    <InfoSection isGraphicLeft={isGraphicLeft}>
+      <Title>{titleText}</Title>
+      <Text>{descriptionText}</Text>
+      <ButtonSection isGraphicLeft={isGraphicLeft}>
+        {buttons.map((button) => (
+          <Link key={button.text} href={button.link}>
+            <a>
+              <SButton backgroundColor={button.color}>{button.text}</SButton>
+            </a>
+          </Link>
+        ))}
+      </ButtonSection>
```
I'd make this an optional prop so something like:

```
{!!isButton && (
<ButtonSection>
...
)}
```
So we can reuse even if a button is not there :')

hirundinidae replied:
Can use without buttons now
![image](https://user-images.githubusercontent.com/65516245/127266871-6c31ef56-1baa-49af-bdbb-ffa3d5968ef2.png)

xsharonhe commented on sections/resources/ResourcesSection.jsx:
```diff
@@ -0,0 +1,91 @@
+        {
+          color: "yellow",
+          text: "Browse Opportunities",
+          link: "/",
```
let's fill in these links here:
direct: ```/resources/opportunities```
digest: ```/resources/blog```
uas: ```/resources/uas```

I forgot to mention we will be removing timeline until the next release (so after August), so could we just have the three?

xsharonhe commented on sections/resources/ResourcesSection.jsx:
```diff
@@ -0,0 +1,91 @@
+        },
+      ],
+      graphic: SampleGraphic,
+      graphicWidth: 500,
+      graphicHeight: 350,
```
instead of this being repeated let's make it a default prop on rhe card so ```graphicWidth = 500``` and so we can save the lines of having to rewrite the same values for graphicWidth and graphicHeight

xsharonhe commented on sections/resources/ResourcesSection.jsx:
```diff
@@ -0,0 +1,91 @@
+      graphic: SampleGraphic,
+      graphicWidth: 500,
+      graphicHeight: 350,
+      isGraphicLeft: true,
```
this is a default prop so not needed when it is ```true```

xsharonhe commented on components/StudentResource.jsx:
```diff
@@ -0,0 +1,124 @@
+`;
+
+const Graphic = styled(Image)`
+  /* TODO: curved border ? */
```
we typically use ```44px``` so feel free to add that now if u want (or later is fine too haha)

hirundinidae replied:
Will add later

xsharonhe reviewed (commented):
Good work! Just general design comments and some updates on my end that I didn't notify you about yet 

One last thing is the buttons are a bit weird when isGraphicLeft is right :
<img width="995" alt="Screen Shot 2021-07-28 at 12 27 40 AM" src="https://user-images.githubusercontent.com/62865150/127264585-9bcb47b3-19b6-431b-a9fc-45f7b04332ff.png">

hirundinidae committed:
Make suggested changes

Fill links
Set default props for image location, width and height
Add ability to use without buttons
Remove timeline resource

xsharonhe reviewed (dismissed):
LGTM

edit: whoops! I think there are still merge conflicts but u can ask for a quick approval at the meeting

hirundinidae committed:
Merge main into add-student-resources
